warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/facet/FacetHandler.java

```diff
@@ -221,118 +200,297 @@ public long process(KEYIN key, RawRecordContainer event, Multimap<String,Normali
         final String shardDate = ShardIdFactory.getDateString(shardId);
         Text dateColumnQualifier = new Text(shardDate);
         
-        HyperLogLogPlus cardinality = new HyperLogLogPlus(10);
-        cardinality.offer(event.getId().toString());
-        
         Text cv = new Text(flatten(event.getVisibility()));
         
-        final HashTableFunction<KEYIN,KEYOUT,VALUEOUT> func = new HashTableFunction<>(contextWriter, context, facetHashTableName, facetHashThreshold,
-                        event.getDate());
-        Multimap<String,NormalizedContentInterface> eventFields = hashEventFields(fields, func);
-        
-        Stream<String> eventFieldKeyStream = eventFields.keySet().stream().filter(new TokenPredicate());
-        if (fieldFilter != null) {
-            eventFieldKeyStream = eventFieldKeyStream.filter(fieldFilter);
+        // filter out event fields that are generated as the result of tokenization.
+        Stream<String> fieldKeyStream = fields.keySet().stream().filter(fieldSelectionPredicate);
+        if (fieldFilterPredicate != null) {
+            fieldKeyStream = fieldKeyStream.filter(fieldFilterPredicate);
         }
-        Set<String> keySet = eventFieldKeyStream.collect(Collectors.toSet());
-        List<Set<String>> keySetList = Lists.newArrayList();
-        keySetList.add(keySet);
-        keySetList.add(keySet);
+        final Set<String> filteredFieldSet = fieldKeyStream.collect(Collectors.toSet());
+        Set<String> pivotFieldSet = new HashSet<>(filteredFieldSet);
+        Set<String> facetFieldSet = new HashSet<>(filteredFieldSet);
+        
+        // fields with a large number of values are hashed. See HashTableFunction for details
+        // @formatter:off
+        final HashTableFunction<KEYIN,KEYOUT,VALUEOUT> func = new HashTableFunction<>(
+                contextWriter, context, facetHashTableName, facetHashThreshold, event.getDate());
+        final Multimap<String,NormalizedContentInterface> eventFields = filterAndHashEventFields(fields, filteredFieldSet, func);
+        // @formatter:on
         
         long countWritten = 0;
         
-        Value sharedValue = new Value(cardinality.getBytes());
-        Multimap<BulkIngestKey,Value> results = ArrayListMultimap.create();
+        // the event id offered to the cardinality is a uid based on the 'EVENT_ID',
+        // so it's helpful to have that around for debugging when logging about the
+        // facet keys that are created.
+        String eventId = null;
```

Comment:
do we still need this? should the use of id below (shard/datatype/uid) replace this?

Reply:
I agree with @FineAndDandy, remove eventId and use the id generated below instead for the debug messages.

Reply:
In this particular case, I needed the identifier that was on the original piece of content that was ingested in order to troubleshoot the problem, not the uid that was being generated because I could not tie that back to the original piece of data e.g via the shard table.